### PR TITLE
updates for x64, visual c++ 2015

### PR DESCRIPTION
**Pull request by dbregman**
1) NAN and INFINITY macros conflict with math.h on some platforms (e.g. visual c++). Use custom macros ECOS_NAN and ECOS_INFINITY as a general fix. IMO this is preferred over trying to tweak includes on various platforms.

2) Fix 64/32 bit confusion. idxint was defined as SuiteSparse_long, but ECOS uses the "int" versions of the SuiteSparse functions. Define idxint to match how ECOS is calling SuiteSparse functions. 

3) set_output_format(_TWO_DIGIT_EXPONENT) call is obsolete (doesn't exist in VC++ 2015), instead of trying to make this conditional for older versions, just remove it since it's cosmetic only.

4) Fix many occurrences of initializing idxint values with "0.0" (which is a double) in the test problems, which caused double-to-int conversion warnings.

5) Create a project file for the test suite (ecostester.c)

6) Do not include windows.h from ecos.h, when including windows.h, use lowercase capitalization to support cross compilers on case sensitive filesystems (MinGW)

7) Fix issues when including from C++ file (extern "C", conflicting "inline" macro, etc)

[Review on Reviewable](https://reviewable.io/reviews/embotech/ecos/122)

### CODE REVIEW

File: src/timer.c [hunk 1]
```diff
@@ -34,16 +34,18 @@
 
 #if (defined WIN32 || _WIN64)
 
+#include <Windows.h>
```
Comment (tkelman) [inline src/timer.c]: Could you make this a lower-case `w`, for the sake of cross-compiling from a case-sensitive OS?

Comment (dbregman) [inline src/timer.c]: The actual filename has a capital W, at least the official one from Microsoft's platform SDK. Your cross-compiler includes might have a version with a lowercase w, but do they all? If so, why are they using different capitalization than Microsoft? It's not obvious to me what's the best thing to do here.

**Comment by adomahidi** [comment]
As far as I know file names on a standard Windows system are case insensitive, so it should work with a lowercase `w`. @dbregman any chance you can confirm and update the PR?

**Comment by dbregman** [comment]
BTW this capitalization thing is not something I changed. Are we sure something is broken here?

**Comment by tkelman** [comment]
I changed it in #113, but that was merged to a different branch. `windows.h` is lower case in every MinGW version I've seen, and MSVC will always be running in a case-insensitive mode.

**Comment by dbregman** [comment]
Thanks, that also made me realize there was a 2nd include of windows.h which happens conditionally (#if CTRLC). Fixed both & Updated.

**Comment by adomahidi** [comment]
Thanks a lot for your effort! It seems that your changes are for Win64, did you test on all platforms, and also Python etc? Heads up: it'll take me some time to review the changes and merge them, but it's on the agenda. Thanks again.

**Comment by echu** [comment]
@adomahidi, any progress on this? would be great to merge these changes for windows.

**Comment by echu** [comment]
Okay. I'm going to deal with this manually. It's been two years...

**Comment by echu** [comment]
Closing this as I manually merged into the develop branch (50b2ced4af816a2285a6fa2c693ba27d5cc19d89).

**Pull request closed without merging**